src/main/java/org/craftercms/studio/impl/v2/service/item/internal/ItemServiceInternalImpl.java

```diff
@@ -413,6 +413,7 @@ public void persistItemAfterCreate(String siteId, String path, String username,
             if (unlock) {
                 item.setState(ItemState.savedAndClosed(item.getState()));
             } else {
+                item.setLockedBy(userObj.getId());
```

Comment:
Shouldn't the current user be the user holding the lock already? Why set it again?

Reply:
When creating an item with "Save" (unlock=false), content is being created as locked since the beginning, but `lockedBy` is not assigned. `lockedBy` being null is what causing issues when trying to unlock afterward.